docs/components/concepts/process-instance-modification.md

```diff
@@ -0,0 +1,226 @@
+when applying the input variable mappings and creating the event subscriptions of the element.
+
+A variable instruction can define the [scope](variables.md#variable-scopes) of the variables. If a scope is defined, the process instance sets the variables as **local** variables in the given scope. For example, set the message variables
+as local variables of the event subprocess. The scope must be a flow scope of the activating element.
+
+If no scope is defined, the process instance sets the variables **globally** in the root scope of the process instance.
+
+## Terminate an element instance
+
+We can use the modification command to terminate an active element instance of the process instance. Consider the
+following example:
+
+![The process instance waits on a task inside a nested interrupting event subprocess. We use the modification command to terminate the event subprocess.](assets/process-instance-modification/process-instance-modification-terminate-element-instance.png)
+
+The process instance completed the first task `A` in the embedded subprocess. It triggered the interrupting timer event
+subprocess and terminated task `B` inside the embedded subprocess. An external system published a message and triggered
+the non-interrupting message boundary event on the subprocess. The process instance waits on task `C` inside the event
+subprocess and on task `D` connected to the boundary event.
+
+The job worker for task `C` failed to complete the job successfully. To skip the task and continue the process instance,
+we modify it and terminate the element instance of the event subprocess.
+
+As a result, the process instance terminates the event subprocess and the element instance of the task `C` that is
+inside the event subprocess. Additionally, the process instance terminates the embedded subprocess because it doesn't
+contain active element instances anymore.
+
+Generally, the modification applies the following rules:
+
+- If the terminating element instance is a subprocess, it terminates all active instances in the subprocess.
+- If the terminating element instance is a call activity, it terminates the child process instance.
+- If the terminating element instance was the last active instance inside a subprocess, it terminates the subprocess.
+- If the terminating element instance was the last active instance of the process instance, it terminates the
+  process instance.
+
+If a terminating element instance is not active, the process instance doesn't apply the termination instruction
+and rejects the command.
+
+:::note
+The process instance can't terminate the last active element instance of a child process instance. As a result, the
+process instance doesn't apply the termination instruction and rejects the command.
+
+Instead, we can terminate the call activity that created the child process instance.
+:::
+
+## Execute the modification instructions
```

Comment:
❓ Do we think this is interesting knowledge for users? I feel like the scenarios have been explained above. The order in which we execute these instructions might be irrelevant.

Reply:
I think that the section is important. It describes two parts:
- the transactional behavior
- the order of the instructions

Both parts are more advanced topics. They are maybe not interesting for users in the beginning. But for advanced users (e.g. users that are aware of the C7 behavior) or as a reference for us when we need to explain the behavior.

For example, to answer the question "Why I don't see a token if I cancel a subprocess and add a new token inside the subprocess?".

@remcowesterhoud does this make sense to you? Should we improve the section?